readme.md

```diff
@@ -21,8 +21,96 @@ text including copyright information.
-https://teeworlds.com/?page=docs&wiki=compiling_everything. In
-particular, you will need SDL2 and FreeType installed.
-
 Originally written by Magnus Auvinen.
+
+
+Building on Linux
+=================
+
+Installing dependencies
+-----------------------
+
+    # Debian/Ubuntu
+    sudo apt install build-essential cmake git libfreetype6-dev libsdl2-dev libpnglite-dev libwavpack-dev python3
+
+    # Fedora
+    sudo dnf install @development-tools cmake gcc-c++ git freetype-devel mesa-libGLU-devel pnglite-devel python3 SDL2-devel wavpack-devel
+
+    # Arch Linux (doesn't have pnglite in its repositories)
+    sudo pacman -S --needed base-devel cmake freetype2 git glu python sdl2 wavpack
+
+
+Downloading repository
+----------------------
+
+    git clone https://github.com/teeworlds/teeworlds --recurse-submodules
+    cd teeworlds
+
+    # If you already cloned the repository before, use:
+    # git submodule update --init
+
+
+Building
+--------
+
+    mkdir -p build
+    cd build
+    cmake ..
+    make
+
+On subsequent builds, you only have to repeat the `make` step.
+
+You can then run the client with `./teeworlds` and the server with
+`./teeworlds_srv`.
+
+
+Build options
+-------------
+
+The following options can be passed to the `cmake ..` command line (between the
+`cmake` and `..`) in the "Building" step above.
+
+`-GNinja`: Use the Ninja build system instead of Make. This automatically
+parallizes the build and is generally **faster**. (Needs `sudo apt install
+ninja-build` on Debian, `sudo dnf install ninja-build` on Fedora, and `sudo
+pacman -S --needed ninja` on Arch Linux.)
+
+`-DDEV=ON`: Enable debug mode and disable some release mechanics. This leads to
+**faster** builds.
+
+`-DCLIENT=OFF`: Disable generation of the client target. Can be useful on
+headless servers which don't have graphics libraries like SDL2 installed.
+
+
+Building on Windows with Visual Studio
+======================================
+
+Download and install some version of [Microsoft Visual
+Studio](https://www.visualstudio.com/) (as of writing, MSVS Community 2017)
+with the following components:
+
+* Desktop development with C++ (on the main page)
+* Python development (on the main page)
+* Git for Windows (in Individual Components → Code tools)
+
+Run Visual Studio. Open the Team Explorer (View → Team Explorer, Ctrl+^,
+Ctrl+M). Click Clone (in the Team Explorer, Connect → Local Git Repositories).
+Enter `https://github.com/teeworlds/teeworlds` into the first input box. Wait
+for the download to complete (terminals might pop up).
+
+Wait until the CMake configuration is done (watch the Output windows at the
+bottom).
+
+Select `teeworlds.exe` in the Select Startup Item… combobox next to the green
+arrow. Wait for the compilation to finish.
+
+For subsequent builds you only have to click the button with the green arrow
+again.
+
+
+Building with bam
+=================
+
+You can also compile Teeworlds with bam, a build system created by matricks.

```

Comment:
Small nitpick, matricks is refered to as "Magnus Auvinen" above. Maybe "a fast and flexible build system" is enough? Other nitpick, it should be made clear that this is valid for all systems, especially since there is no "Building on Mac" section.

Reply:
Adjusted.